### PR TITLE
DOC-1397:  Connect - Initial docs

### PR DESCRIPTION
**Pull request by lanceleonard**
- Initial draft

### CODE REVIEW

**Comment by lanceleonard** [comment]
Staged previews [are available](https://docs.redis.com/staging/jira-doc-1397/connect/)

File: content/connect/upgrade.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+    ``` console
+    rladmin upgrade module db_name redis-connect module_name rg version 999999 module_args keep_args
+    ```
+
+    The passed with the `rg version` parameter is the version number associated with the RedisGears update.  You can find this in the JSON file provided with your download.
+
+    Be sure to specify the `keep_args` parameter; otherwise, the configuration will be reset, with unpredictable results.
+
+    To verify the RedisGears version used with your RedisConnect instance:
```
Comment (rrelledge) [inline content/connect/upgrade.md:87]:
```suggestion
    To verify the RedisGears version used with your Redis Connect instance:
```

File: content/connect/architecture.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+linkTitle: Architecture & components
+description: Describes the architecture and components of Redis Connect
+weight: $weight
+alwaysopen: false
+categories: ["Connect"]
+aliases: /connect/architecture/
+         /connect/architecture.md
+---
+
+Redis Connect relies on a variety components, as shown in the following figure:
```
Comment (rrelledge) [inline content/connect/architecture.md]:
```suggestion
Redis Connect relies on a variety of components, as shown in the following figure:
```

File: content/connect/architecture.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+{{<image filename="images/connect/redis-connect-architecture.png" alt="Redis Connect architecture and components" width="75%">}}{{< /image >}}
+
+For Preview, Redis Connect uses [Debezium](https://debezium.io/) to load source data into a Redis Connect instance as [Redis Streams](https://redis.io/docs/manual/data-types/streams/).
+
+Currently, Redis Connect uses the [Debezium Redis Sink Connector](https://debezium.io/documentation/reference/stable/operations/debezium-server.html#_redis_stream).  Future plans include support for additional connectors and a related SDK.
+
+Once data in loaded into the Redis Connect instance, [Redis Gears recipes](https://developer.redis.com/howtos/redisgears/) transform it into a format appropriate to the target Redis database.
```
Comment (rrelledge) [inline content/connect/architecture.md:20]:
```suggestion
Once data is loaded into the Redis Connect instance, [RedisGears recipes](https://developer.redis.com/howtos/redisgears/) transform it into a format appropriate for the target Redis database.
```

Comment (rrelledge) [inline content/connect/architecture.md:20]:
```suggestion
Once data in loaded into the Redis Connect instance, [RedisGears recipes](https://developer.redis.com/howtos/redisgears/) transform it into a format appropriate to the target Redis database.
```

File: content/connect/architecture.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+Currently, incoming database rows are transformed into [Redis hashes](https://redis.io/docs/manual/data-types/#hashes) that incorporate the original table name into key names.  Future updates will add additional transformations, such as [RedisJSON documents](https://redis.io/docs/stack/json/).
+
+## Data transformation engine
+
+The Redis Connect data transformation engine uses Redis Gears to perform two major functions:
```
Comment (rrelledge) [inline content/connect/architecture.md:26]:
```suggestion
The Redis Connect data transformation engine uses RedisGears to perform two major functions:
```

File: content/connect/install.md [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+
+Once Redis Connect CLI is installed, you can use it to create a new Redis Connect instance.  Here's how:
+
+1.  Verify prerequisites
+
+    To set up a new Redis Connect instance, you need:
+
+    - A Redis Enterprise cluster, v6.2 or later
+
+    - [Redis Gears](https://docs.redis.com/latest/modules/redisgears/) v1.2.4 or later must be installed on the cluster.
```
Comment (rrelledge) [inline content/connect/install.md:49]:
```suggestion
    - [RedisGears](https://docs.redis.com/latest/modules/redisgears/) v1.2.4 or later must be installed on the cluster
```

File: content/connect/install.md [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+
+1.  Use redis-connect CLI to create a new Redis Connect instance on your Redis Enterprise cluster
+
+    ``` console
+    redis-connect create
+    ```
+
+    This creates a database named `redis-connect` in your cluster
+    
+    Your user account needs permission to create databases and to register Redis Gears recipes
```
Comment (rrelledge) [inline content/connect/install.md:61]:
```suggestion
    Your user account needs permission to create databases and to register RedisGears recipes
```

File: content/connect/architecture.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+Redis Connect requires the following credentials and secrets:
+
+- The Redis Connect Data Transformation Engine requires the credentials and certificate to connect and write encrypted data to the target Redis database
+
+- The Debezium Redis Sink Connector requires the credentials and certificate to connect and write data into the Redis Connect database
+
+- The Debezium Source Connector requires read access to the source database, including related secrets
+
+- Redis Connect CLI requires a certificates needed to connect to the Redis Enterprise cluster
```
Comment (rrelledge) [inline content/connect/architecture.md:48]:
```suggestion
- Redis Connect CLI requires any certificates needed to connect to the Redis Enterprise cluster 
```

File: content/connect/architecture.md [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+
+- The Redis Connect Data Transformation Engine requires the credentials and certificate to connect and write encrypted data to the target Redis database
+
+- The Debezium Redis Sink Connector requires the credentials and certificate to connect and write data into the Redis Connect database
+
+- The Debezium Source Connector requires read access to the source database, including related secrets
+
+- Redis Connect CLI requires a certificates needed to connect to the Redis Enterprise cluster 
+
+    In order to run the `create` command, Redis Connect CLI the credentials of a privileged user.  These credentials are not cached or stored
```
Comment (rrelledge) [inline content/connect/architecture.md:50]:
```suggestion
    In order to run the `create` command, Redis Connect CLI requires the credentials of a privileged user.  These credentials are not cached or stored.
```

File: content/connect/architecture.md [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+- Store them as environment variables
+
+- Leverage a key management service, such as Google Cloud [Key Management](https://cloud.google.com/security-key-management), Microsoft Azure [Key Vault](https://docs.microsoft.com/en-us/azure/key-vault/), [HashiCorp Vault](https://www.hashicorp.com/products/vault), [and others](https://en.wikipedia.org/wiki/Key_management)
+  
+
+## Scalability and availability
+
+Redis Connect is highly available:
+
+- Debezium Server is deployed using [Kubernetes](https://kubernetes.io/) or [Pacemaker](https://clusterlabs.org/pacemaker/), when enables failover between stateless instances while securing state information in Redis
```
Comment (rrelledge) [inline content/connect/architecture.md:65]:
```suggestion
- Debezium Server is deployed using [Kubernetes](https://kubernetes.io/) or [Pacemaker](https://clusterlabs.org/pacemaker/), which enables failover between stateless instances while securing state information in Redis
```

File: content/connect/architecture.md [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+
+- Redis Connect relies on Redis Enterprise high availability mechanisms, such as replica shards, cluster configuration, and so on
+
+- Horizontal scalability is handled by splitting incoming data streams across multiple shards and writing updates in parallel.  Original table order is retained
```
Comment (rrelledge) [inline content/connect/architecture.md:69]:
```suggestion
- Horizontal scalability is handled by splitting incoming data streams across multiple shards and writing updates in parallel.  The original table order is retained.
```

File: content/connect/architecture.md [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+
+
+## Tested topologies
+
+Redis Connect runs on Redis Enterprise, which means it works on any deployment, whether "bare metal," virtual machines (VMs), or containers.  Redis Connect can be run on the same cluster as the target Redis database or update a database stored on a separate cluster.
```
Comment (rrelledge) [inline content/connect/architecture.md:74]:
```suggestion
Redis Connect runs on Redis Enterprise, which means it works on any deployment, whether "bare metal," virtual machines (VMs), or containers.  Redis Connect can run on the same cluster as the target Redis database or update a database stored on a separate cluster.
```

File: content/connect/architecture.md [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+
+Debezium Server should run in "warm standby" [high availability](https://en.wikipedia.org/wiki/High_availability_software).  We recommend one of the following:
+
+- Run using two VMs in an active-passive configuration.  /
```
Comment (rrelledge) [inline content/connect/architecture.md:78]:
```suggestion
- Run using two VMs in an active-passive configuration.  
```

File: content/connect/architecture.md [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+
+    Failover should be orchestrated by a high availability resource manager, such as  [Pacemaker](https://clusterlabs.org/pacemaker/doc/)
+
+- Run as a [Kubernetes pod](https://en.wikipedia.org/wiki/Kubernetes#Pods) where Kubernetes monitors pod activity (liveliness and readiness) and recovers non-functioning pods
+
+    Note that Redis Connect stores state information in the Redis Connect instance.  No state information is stored at the connector level
```
Comment (rrelledge) [inline content/connect/architecture.md:84]:
```suggestion
    Note that Redis Connect stores state information in the Redis Connect instance.  No state information is stored at the connector level.
```

File: content/connect/install.md [hunk 14]
```diff
@@ -0,0 +1,258 @@
...
+weight: $weight
+alwaysopen: false
+categories: ["Connect"]
+aliases: /connect/install/
+         /connect/install.md
+---
+
+Redis Connect helps connect traditional relational database systems (RDBMS) to Redis Enterprise.  It does this by ingesting (_importing_) row-based data into a target Redis database.
+
+The Redis Connect command-line interface (CLI) helps you set up a new Redis Connect instance and to configure the connections between your source data and Redis Enterprise.
```
Comment (rrelledge) [inline content/connect/install.md:14]:
```suggestion
The Redis Connect command-line interface (CLI) helps you set up a new Redis Connect instance and configure the connections between your source data and Redis Enterprise.
```

File: content/connect/install.md [hunk 15]
```diff
@@ -0,0 +1,258 @@
...
+
+## Install Redis Connect CLI
+
+Redis Connect CLI requires Python 3.7 or later.  If your operating system relies on a different version of Python, we recommend installing the latest version of Python and using [virtual environments](https://docs.python.org/3/library/venv.html) to manage Python.
+
+Specific steps vary according to your operating system and environment.  The general process is:
+
+1.  Install or update to [Python 3.7 or later](https://www.python.org/downloads/)
+
+1.  Set up and activate a new virtual environment for Redis Connect
```
Comment (rrelledge) [inline content/connect/install.md:26]:
```suggestion
1.  Set up and activate a new virtual environment for Redis Connect:
```

File: content/connect/install.md [hunk 16]
```diff
@@ -0,0 +1,258 @@
...
+
+    ``` console
+    python3 -m venv venv
+    source venv/bin/activate
+    ```
+
+1.  Install Redis Connect CLI.
```
Comment (rrelledge) [inline content/connect/install.md:33]:
```suggestion
1.  Install Redis Connect CLI:
```

File: content/connect/install.md [hunk 17]
```diff
@@ -0,0 +1,258 @@
...
+
+1.  Verify prerequisites
+
+    To set up a new Redis Connect instance, you need:
+
+    - A Redis Enterprise cluster, v6.2 or later
+
+    - [Redis Gears](https://docs.redis.com/latest/modules/redisgears/) v1.2.4 or later must be installed on the cluster.
+
+    - A target Redis database, which can be added after setting up Redis Connect.
```
Comment (rrelledge) [inline content/connect/install.md:51]:
```suggestion
    - A target Redis database, which can be added after setting up Redis Connect
```

File: content/connect/install.md [hunk 18]
```diff
@@ -0,0 +1,258 @@
...
+
+1.  Use redis-connect CLI to create a new Redis Connect instance on your Redis Enterprise cluster
```
Comment (rrelledge) [inline content/connect/install.md:53]:
```suggestion
1.  Use `redis-connect` CLI to create a new Redis Connect instance on your Redis Enterprise cluster:
```

File: content/connect/install.md [hunk 19]
```diff
@@ -0,0 +1,258 @@
...
+    
+    Your user account needs permission to create databases and to register Redis Gears recipes
+
+1.  Use the `scaffold` command to create the configuration files for your Redis Connect instance
+
+    ``` console
+    redis-connect scaffold <PROJECT_NAME>
+    ```
+
+    This create a directory in the current location; this directory contains the following files:
```
Comment (rrelledge) [inline content/connect/install.md:69]:
```suggestion
    This creates a directory in the current location; this directory contains the following files:
```

File: content/connect/install.md [hunk 20]
```diff
@@ -0,0 +1,258 @@
...
+
+    ```bash
+    ├── debezium
+    │   └── application.properties
+    └── config.yaml
+    ```
+
+1.  Update the configuration files
+
+    1.  The Redis Connect configuration file (`config.yaml`) file defines the connection between your source data and your Redis database
```
Comment (rrelledge) [inline content/connect/install.md:79]:
```suggestion
    1.  The Redis Connect configuration file (`config.yaml`) defines the connection between your source data and your Redis database 
```

File: content/connect/install.md [hunk 21]
```diff
@@ -0,0 +1,258 @@
...
+
+1.  Deploy Redis Connect configuration
+
+    The `deploy` command deploys a Redis Connect configuration to your Redis database instance:
+
+    ``` console
+    redis-connect deploy
+    ```
+
+    The Redis Connect configuration files is saved (_persisted_) to the cluster, which helps configure new shards and recover in the event of shard or node failure.
```
Comment (rrelledge) [inline content/connect/install.md:123]:
```suggestion
    The Redis Connect configuration file is saved (_persisted_) to the cluster, which helps configure new shards and recover in the event of shard or node failure.
```

File: content/connect/install.md [hunk 22]
```diff
@@ -0,0 +1,258 @@
...
+
+ Performance statistics per batch (batch size: 100):
+  Last run duration (ms): 0
+  Average run duration (ms): 0
+  Last estimated lag (ms): 0
+  Average estimated lag (ms): 0
+```
+    {{%/expand%}}
+
+    If you receive a `No streams found` error, make sure that Debezium server is properly set up and connected to your source database.
```
Comment (rrelledge) [inline content/connect/install.md:163]:
```suggestion
    If you receive a `No streams found` error, make sure that the Debezium server is properly set up and connected to your source database.
```

File: content/connect/install.md [hunk 23]
```diff
@@ -0,0 +1,258 @@
...
+    
+    For help, see [Install Debezium Server](#install-debezium-server).
+
+## Install Debezium Server
+
+The follow steps provide a high level overview to the Debezium setup process.  For a detailed look, see the [Debezium docs](https://debezium.io/documentation/).
```
Comment (rrelledge) [inline content/connect/install.md:169]:
```suggestion
The follow steps provide a high level overview of the Debezium setup process.  For a detailed look, see the [Debezium docs](https://debezium.io/documentation/).
```

File: content/connect/install.md [hunk 24]
```diff
@@ -0,0 +1,258 @@
...
+
+1.  _(Optional)_ Set up an example database
+
+    The Debezium project includes preconfigured examples for many relational database.
```
Comment (rrelledge) [inline content/connect/install.md:173]:
```suggestion
    The Debezium project includes preconfigured examples for many relational databases.
```

File: content/connect/install.md [hunk 25]
```diff
@@ -0,0 +1,258 @@
...
+
+    ```bash
+    sudo podman run -it --rm --name debezium --network=host --privileged -v $PWD/debezium:/debezium/conf debezium/server
+    ```
+
+## Resolve Debezium Oracle errors
+
+If you are using Oracle as your source database, be aware that Debezium Server does _not_ include the Oracle Debezium Connector or the related JDBC driver.
+
+As result, Debezium fails with an error.  If this happens:
```
Comment (rrelledge) [inline content/connect/install.md:245]:
```suggestion
As a result, Debezium fails with an error.  If this happens:
```

File: content/connect/install.md [hunk 26]
```diff
@@ -0,0 +1,258 @@
...
+
+1.  Add the connector and the JDBC driver
+2.  Restart Debezium Server
+
+The following example shows one way to do this
```
Comment (rrelledge) [inline content/connect/install.md:250]:
```suggestion
The following example shows one way to do this:
```

File: content/connect/upgrade.md [hunk 27]
```diff
@@ -0,0 +1,101 @@
...
+
+These steps provide more details:
+
+1. Download the most recent version of RedisGears
+
+    ``` console
+    curl -s http://redismodules.s3.amazonaws.com/redisgears/snapshots/redisgears_python.Linux-ubuntu18.04-x86_64.master.zip -o /tmp/redisgears.zip
+    ```
+
+2.  Use the `v2/modules/` REST endpoint to install the RedisGears update in your cluster
```
Comment (rrelledge) [inline content/connect/upgrade.md:56]:
```suggestion
2.  Use the `POST /v2/modules/` REST API request to install the RedisGears update in your cluster
```

File: content/connect/upgrade.md [hunk 28]
```diff
@@ -0,0 +1,101 @@
...
+    ```
+    The response object includes an action ID to help track installation progress  
+    
+    Use the `/actions/` endpoint to determine installation progress:
+
+    ``` console
+    curl -k -s -u "<CLUSTER_USER>:<CLUSTER_PASSWORD>" https://<CLUSTER_HOST>:<CLUSTER_API_PORT>/v1/actions/${action_uid}
+    ```
+
+    This endpoint returns `complete` when installation is finished.
```
Comment (rrelledge) [inline content/connect/upgrade.md:69]:
```suggestion
    This request returns `complete` when installation is finished.
```

File: content/connect/upgrade.md [hunk 29]
```diff
@@ -0,0 +1,101 @@
...
+
+3.  Update the Redis Connect instance to use the RedisGears update
+    
+    Use `rladmin` to update your Redis Connect instance to the correct RedisGears version:
+
+    ``` console
+    rladmin upgrade module db_name redis-connect module_name rg version 999999 module_args keep_args
+    ```
+
+    The passed with the `rg version` parameter is the version number associated with the RedisGears update.  You can find this in the JSON file provided with your download.
```
Comment (rrelledge) [inline content/connect/upgrade.md:83]:
```suggestion
    The `rg version` parameter is the version number associated with the RedisGears update.  You can find this in the JSON file provided with your download.
```

File: content/connect/upgrade.md [hunk 30]
```diff
@@ -0,0 +1,101 @@
...
+
+    To verify the RedisGears version used with your RedisConnect instance:
+    
+    1.  Select the **Databases** menu of the admin console and then select redis-connect from the list of databases.  Select the **Configuration** tab and then locate the **Redis modules** section.
+        
+        If an information icon appears to the right of the RedisGears version, the update is available for use.
+
+        {{<image filename="images/connect/redis-connect-db-update-available.png" alt="When module updates are avilable, an information icon appears to the right of the current version displayed in the database configuration." width="75%">}}{{< /image >}}
+        
+    2.  You can also also use `rladmin` to display the RedisGears version for your Redis Connect instance:
```
Comment (rrelledge) [inline content/connect/upgrade.md:95]:
```suggestion
    2.  You can also use `rladmin` to display the RedisGears version for your Redis Connect instance:
```

**Review by rrelledge — APPROVED** [review]
Looks good to me. I left some suggestions for typos and such.

File: content/connect/upgrade.md (hunk 1 above)
Comment (lanceleonard) [reply content/connect/upgrade.md:87]: Fixed

File: content/connect/architecture.md (hunk 2 above)
Comment (lanceleonard) [reply content/connect/architecture.md]: Fixed.

File: content/connect/architecture.md (hunk 4 above)
Comment (lanceleonard) [reply content/connect/architecture.md:26]: Fixed.

File: content/connect/install.md (hunk 5 above)
Comment (lanceleonard) [reply content/connect/install.md:49]: Fixed.

File: content/connect/install.md (hunk 26 above)
Comment (lanceleonard) [reply content/connect/install.md:250]: Fixed

File: content/connect/upgrade.md (hunk 28 above)
Comment (lanceleonard) [reply content/connect/upgrade.md:69]: Fixed

File: content/connect/upgrade.md (hunk 29 above)
Comment (lanceleonard) [reply content/connect/upgrade.md:83]: Fixed.

File: content/connect/upgrade.md (hunk 30 above)
Comment (lanceleonard) [reply content/connect/upgrade.md:95]: Fixed.

File: content/connect/install.md (hunk 6 above)
Comment (lanceleonard) [reply content/connect/install.md:61]: Fixed.

File: content/connect/architecture.md (hunk 7 above)
Comment (lanceleonard) [reply content/connect/architecture.md:48]: Fixed

File: content/connect/architecture.md (hunk 8 above)
Comment (lanceleonard) [reply content/connect/architecture.md:50]: Fixed.

File: content/connect/architecture.md (hunk 9 above)
Comment (lanceleonard) [reply content/connect/architecture.md:65]: Fixed.

File: content/connect/architecture.md (hunk 10 above)
Comment (lanceleonard) [reply content/connect/architecture.md:69]: Fixed.

File: content/connect/architecture.md (hunk 11 above)
Comment (lanceleonard) [reply content/connect/architecture.md:74]: Fixed.

File: content/connect/install.md (hunk 15 above)
Comment (lanceleonard) [reply content/connect/install.md:26]: Fixed.

File: content/connect/install.md (hunk 17 above)
Comment (lanceleonard) [reply content/connect/install.md:51]: Fixed

File: content/connect/install.md (hunk 18 above)
Comment (lanceleonard) [reply content/connect/install.md:53]: Fixed.

File: content/connect/install.md (hunk 20 above)
Comment (lanceleonard) [reply content/connect/install.md:79]: Fixed.

File: content/connect/install.md (hunk 21 above)
Comment (lanceleonard) [reply content/connect/install.md:123]: Fixed.

File: content/connect/install.md (hunk 22 above)
Comment (lanceleonard) [reply content/connect/install.md:163]: Fixed.

File: content/connect/install.md (hunk 23 above)
Comment (lanceleonard) [reply content/connect/install.md:169]: Fixed.

File: content/connect/install.md (hunk 24 above)
Comment (lanceleonard) [reply content/connect/install.md:173]: Fixed.

File: content/connect/install.md (hunk 25 above)
Comment (lanceleonard) [reply content/connect/install.md:245]: Fixed.

**Pull request merged**